enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -52,6 +52,10 @@ def start_kernel_from_session(self, kernel_id, kernel_name, connection_info, pro
         # Load connection info into member vars - no need to write out connection file
         km.load_connection_info(connection_info)
 
+        info_key = connection_info.get('key')
+        if info_key:
+            connection_info['key'] = bytes_to_str(info_key)
+
```

Comment:
This looks fine. I think we should add a comment as to why this is necessary since `load_connection_info()` is doing the exact opposite operation (converting the key to bytes). As a result, someone looking at this will find it odd. I think something like `Convert key back to string so json operations are properly behaved` may be sufficient.

Reply:
@esevan - do you mind giving this a look as well?  Thanks.

Reply:
I'm the one confused by missing comment :confused:. Looks good to me except for the comment thing :+1: